crane-cloud/frontend
# feat: implement mockups for MLflow interface for notebook experiments

# Description

- This PR introduces a guiding interface to integrate the MLOps lifecycle for managing Jupyter notebook experiments runs with MLflow.

## Type of change


## Clickup Ticket ID

https://app.clickup.com/t/8696k7d7r

## How Can This Been Tested?

- Run this branch locally and checkout any of your Jupyter notebooks or create one! 
- In the app deployment dashboard, you will see new elements in the sidebar specific for this deployment (Notebooks) i.e. `Experiments` and `AI Deployments`
- When you click around on the experiments sections, you should be able to do the following:
1. A button to create a new experiment
2. A filter section for runs created and their status
3. A tabular section for experiment runs from your notebook
4. Checkboxes to delete one or more experiment runs
5. Clicking on a row should take you to an experiment runs' details page.

NB: include this in your environment variables - `REACT_APP_MLOPS_API_URL="https://mlops.renu-01.cranecloud.io"`


## Checklist:


# Screenshots

![Screenshot 2024-11-20 at 11 48 35 PM](https://github.com/user-attachments/assets/7d0e73e3-8533-4eb5-a176-1edb951809f8)

![Screenshot 2024-11-20 at 11 47 01 PM](https://github.com/user-attachments/assets/dbd8f15e-9e8f-4a7e-a1c6-6d1b0a60703e)

![Screenshot 2024-11-20 at 11 47 57 PM](https://github.com/user-attachments/assets/1c488627-4de9-4ad0-be89-99805da36785)

![Screenshot 2024-11-20 at 11 49 26 PM](https://github.com/user-attachments/assets/583d97c9-8f5d-4dbf-93e7-5757b05fdbe7)

![Screenshot 2024-11-20 at 11 49 45 PM](https://github.com/user-attachments/assets/1beb235f-5968-4e71-ad94-6203333e12ed)
